delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/resources/ResourceEvent.scala

```diff
@@ -0,0 +1,170 @@
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary.{contexts, nxv}
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.ContextValue
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.JsonLdContext.keywords
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.{CompactedJsonLd, ExpandedJsonLd, JsonLdEncoder}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.Subject
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{BaseUri, Event, ResourceRef}
+import io.circe.generic.extras.Configuration
+import io.circe.generic.extras.semiauto.deriveConfiguredEncoder
+import io.circe.{Encoder, Json}
+
+import scala.annotation.nowarn
+
+/**
+  * Enumeration of resource event states
+  */
+sealed trait ResourceEvent extends Event {
+
+  /**
+    * @return the resource identifier
+    */
+  def id: Iri
+
+  /**
+    * @return the project where the resource belongs to
+    */
+  def project: ProjectRef
+
+  /**
+    * @return the collection of known resource types
+    */
+  def types: Set[Iri]
+
+}
+
+object ResourceEvent {
+
+  /**
+    * Event representing a resource creation.
+    *
+    * @param id          the resource identifier
+    * @param project     the project where the resource belongs
+    * @param schema      the schema used to constrain the resource
+    * @param types       the collection of known resource types
+    * @param source      the representation of the resource as posted by the subject
+    * @param compacted   the compacted JSON-LD representation of the resource
+    * @param expanded    the expanded JSON-LD representation of the resource
+    * @param rev         the resource revision
+    * @param instant     the instant when this event was created
+    * @param subject     the subject which created this event
+    */
+  final case class ResourceCreated(
+      id: Iri,
+      project: ProjectRef,
+      schema: ResourceRef,
+      types: Set[Iri],
+      source: Json,
+      compacted: CompactedJsonLd,
+      expanded: ExpandedJsonLd,
+      rev: Long,
+      instant: Instant,
+      subject: Subject
+  ) extends ResourceEvent
+
+  /**
+    * Event representing a resource modification.
+    *
+    * @param id          the resource identifier
+    * @param project     the project where the resource belongs
+    * @param types       the collection of known resource types
+    * @param source      the representation of the resource as posted by the subject
+    * @param compacted   the compacted JSON-LD representation of the resource
+    * @param expanded    the expanded JSON-LD representation of the resource
+    * @param rev         the resource revision
+    * @param instant     the instant when this event was created
+    * @param subject     the subject which created this event
+    */
+  final case class ResourceUpdated(
+      id: Iri,
+      project: ProjectRef,
```

Comment:
We can't update the schema ?
If we can, we could have 2 types of events no ?

Reply:
Once the resource has been created against one schema it cannot be updated against another schema.